### PR TITLE
msg/async/rdma: fixup useless code

### PR DESCRIPTION
**Pull request by gongchuang**
Signed-off-by: gongchuang <gong.chuang@zte.com.cn>

### CODE REVIEW

**Comment by gongchuang** [comment]
retest this please

File: src/msg/async/rdma/RDMAServerSocketImpl.cc [hunk 1]
```diff
@@ -26,10 +26,9 @@ int RDMAServerSocketImpl::listen(entity_addr_t &sa, const SocketOptions &opt)
   int rc = 0;
   server_setup_socket = net.create_socket(sa.get_family(), true);
   if (server_setup_socket < 0) {
-    rc = -errno;
     lderr(cct) << __func__ << " failed to create server socket: "
                << cpp_strerror(errno) << dendl;
-    return rc;
+    return -errno;
```
Comment (liewegas) [inline src/msg/async/rdma/RDMAServerSocketImpl.cc]: actually you should change cpp_strerror to use rc instead.  the lderr op might make a syscall and reset errno.

File: src/msg/async/rdma/RDMAServerSocketImpl.cc [hunk 2]
```diff
@@ -45,15 +44,13 @@ int RDMAServerSocketImpl::listen(entity_addr_t &sa, const SocketOptions &opt)
 
   rc = ::bind(server_setup_socket, sa.get_sockaddr(), sa.get_sockaddr_len());
   if (rc < 0) {
-    rc = -errno;
     ldout(cct, 10) << __func__ << " unable to bind to " << sa.get_sockaddr()
                    << " on port " << sa.get_port() << ": " << cpp_strerror(errno) << dendl;
```
Comment (liewegas) [inline src/msg/async/rdma/RDMAServerSocketImpl.cc]: here too, use rc for cpp_strerror

File: src/msg/async/rdma/RDMAServerSocketImpl.cc [hunk 3]
```diff
@@ -45,15 +44,13 @@ int RDMAServerSocketImpl::listen(entity_addr_t &sa, const SocketOptions &opt)
...
   if (rc < 0) {
-    rc = -errno;
     ldout(cct, 10) << __func__ << " unable to bind to " << sa.get_sockaddr()
                    << " on port " << sa.get_port() << ": " << cpp_strerror(errno) << dendl;
     goto err;
   }
 
   rc = ::listen(server_setup_socket, 128);
   if (rc < 0) {
-    rc = -errno;
```
Comment (liewegas) [inline src/msg/async/rdma/RDMAServerSocketImpl.cc]: and here

**Comment by gongchuang** [comment]
@liewegas ok

File: src/msg/async/rdma/RDMAServerSocketImpl.cc [hunk 4]
```diff
@@ -28,7 +28,7 @@ int RDMAServerSocketImpl::listen(entity_addr_t &sa, const SocketOptions &opt)
   if (server_setup_socket < 0) {
```
Comment (yuyuyu101) [inline src/msg/async/rdma/RDMAServerSocketImpl.cc]: yes, create_socket will return -errno, let's change to this

Comment (gongchuang) [inline src/msg/async/rdma/RDMAServerSocketImpl.cc]: modify as the following：set “rc = server_setup_socket;” ？
if (server_setup_socket < 0) {
    rc = server_setup_socket;
    lderr(cct) << __func__ << " failed to create server socket: "
               << cpp_strerror(rc) << dendl;
    return rc;
  }

**Comment by liewegas** [comment]
needs rebase

**Comment by yuyuyu101** [comment]
needs rebase

**Comment by tchaikov** [comment]
@gongchuang  ping?

**Comment by yuyuyu101** [comment]
feel free to rebase and reopen

**Pull request closed without merging**